apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx

```diff
@@ -77,8 +80,10 @@ const AnalyticsApp = (props: Props) => {
 
   const metricName = runReportResponse.metricHeaders && runReportResponse.metricHeaders[0].name;
 
+  const pendingData = isEmpty(runReportResponse) && !error && runReportFetchRequirements;
```

Comment:
this helps resolve the gap between when our artificial 'loading' state is done and when the data has actually been gathered.